### PR TITLE
Support duplicated mode in DynamicFragment

### PR DESCRIPTION
**Pull request by acezen**
## What do these changes do?
- add a new loading graph mode which every worker duplicated load the whole graph.
- make duplicated load graph mode as default of networkx.
- use num_workers=2 in nx test

 DynamicFragment now support two type of graph storage mode.
 - distributed mode: the storage is like other Fragments that every fragment holds a part of graph.
 - duplicated mode: every fragment holds the whole graph which duplicated in workers (the mode is designed for APSP-like algorithms and default in networkx).
 
if users want to use distributed mode to load graph, just init graph by

```python
G = nx.Graph(dist=True)
```

## Related issue number
#317 

## TODO:
- [x] pass the test case for default duplicated load.
- [ ] ~~add new test case to check the add/delete/update is correct on all workers.~~  (move to #336 )

### CODE REVIEW

**Comment by yecol** [comment]
maybe revise the parameter name.
```
G = nx.Graph(duplicated_load=True)
```

File: analytical_engine/core/vertex_map/dynamic_fragment_vertex_map.h [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  void Init(fid_t fnum) override {
+    Base::Init(fnum);
+    duplicated_load_ = true;
+  }
+
+  void Construct() override {
+    if (!duplicated_load_) {
+      // only the multiple fragments vertex map need to construct.
+      Base::Construct();
+    }
```
Comment (sighingnow) [inline analytical_engine/core/vertex_map/dynamic_fragment_vertex_map.h:51]: Otherwise we don't need a vertexmap?

**Review by sighingnow — APPROVED** [review]

**Review by pwrliang — CHANGES_REQUESTED** [review]

File: analytical_engine/core/vertex_map/dynamic_fragment_vertex_map.h [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+template <typename OID_T, typename VID_T>
+class DynamicFragmentVertexMap : public grape::GlobalVertexMap<OID_T, VID_T> {
+  using Base = grape::GlobalVertexMap<OID_T, VID_T>;
+
+ public:
+  explicit DynamicFragmentVertexMap(const grape::CommSpec& comm_spec)
+      : Base(comm_spec), duplicated_load_(false) {}
+  ~DynamicFragmentVertexMap() = default;
+
+  void Init() override { Base::Init(); }
```
[the remark this answers is not in the document]
Comment (acezen) [reply analytical_engine/core/vertex_map/dynamic_fragment_vertex_map.h:40]: good suggestion, that would be more clear, I would fix this.

[the remark this answers is not in the document]
Comment (acezen) [reply analytical_engine/core/vertex_map/dynamic_fragment_vertex_map.h]: fixed.

**Review by pwrliang — APPROVED** [review]

**Comment by yecol** [comment]
according to our discussion: 
```python
# by default, duplicated load
G = nx.Graph()

# or load a fragment, rather than a whole graph.
G = nx.Graph(distributed_load=True)
```

**Comment by acezen** [comment]
hi, @yecol @sighingnow @pwrliang the refactoring of duplicated load of DynamicFragment is done. the pr needs you to review again. Thanks a lot.

**Review by yecol — APPROVED** [review]

**Review by sighingnow — APPROVED** [review]

**Pull request merged**